axum/src/typed_header.rs

```diff
@@ -1,20 +1,20 @@
-use super::{FromRequest, RequestParts};
+use crate::extract::{FromRequest, RequestParts};
 use async_trait::async_trait;
 use axum_core::response::{IntoResponse, IntoResponseParts, Response, ResponseParts};
 use headers::HeaderMapExt;
 use http::header::{HeaderName, HeaderValue};
 use std::ops::Deref;
 
-/// Extractor that extracts a typed header value from [`headers`].
+/// Extractor and response that works with typed header values from [`headers`].
+///
+/// # As extractor
 ///
 /// In general, it's recommended to extract only the needed headers via `TypedHeader` rather than
 /// removing all headers with the `HeaderMap` extractor.
 ///
-/// # Example
-///
 /// ```rust,no_run
 /// use axum::{
-///     extract::TypedHeader,
+///     TypedHeader,
```

Comment:
I think oder rustfmt formatting rules this should now go on the same line as `Router` below. There's a nightly setting you can use to have rustfmt format doc codeblocks.

Reply:
Ah nice! I didn't realize. I'll probably do another PR that formats the docs and merges imports. Not sure if that feature is stable in rustfmt yet.